deepmd/train/trainer.py

```diff
@@ -269,21 +269,48 @@ def fitting_net_init(fitting_type_, descrpt_type_, params):
             )
 
         # learning rate
-        lr_param = j_must_have(jdata, "learning_rate")
-        scale_by_worker = lr_param.get("scale_by_worker", "linear")
-        if scale_by_worker == "linear":
-            self.scale_lr_coef = float(self.run_opt.world_size)
-        elif scale_by_worker == "sqrt":
-            self.scale_lr_coef = np.sqrt(self.run_opt.world_size).real
-        else:
-            self.scale_lr_coef = 1.0
-        lr_type = lr_param.get("type", "exp")
-        if lr_type == "exp":
-            self.lr = LearningRateExp(
-                lr_param["start_lr"], lr_param["stop_lr"], lr_param["decay_steps"]
-            )
+        if not self.multi_task_mode:
```

Comment:
Here a lot of codes are duplicated. I suggest creating a new method that returns lr and coef to merge duplicated codes.